src/API/Google/Proxy.php

```diff
@@ -173,6 +175,10 @@ protected function create_merchant_account_request( string $name, string $site_u
 				throw InvalidTerm::contains_invalid_terms( $name );
 			}
 
+			if ( preg_match( '/URL?.* invalid top-level domain name/', $message ) ) {
+				throw InvalidDomainName::invalid_top_level_domain_name( $this->get_site_url() );
```

Comment:
Would it look a bit cleaner to pass the URL through `strip_url_protocol`, it's what we used in other places where we expose the domain name in the UI?